### PR TITLE
add new module for webapp slot

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

add new module for webapp slot

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yungezz** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@zikalino @nitzmahone @jborean93 Could you help review this PR and push to merged?

File: lib/ansible/modules/cloud/azure/azure_rm_webappslot.py [hunk 1]
```diff
@@ -0,0 +1,1039 @@
...
+            - Set it target slot name to auto swap.
+            - Set it to False to disable auto slot swap.
+    swap:
+        description:
+            - Swap deployment slots of a web app.
+        suboptions:
+            action:
+                description:
+                    - Swap types.
+                    - preview is to apply target slot settings on source source slot first.
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_webappslot.py]: s/source source/source/

File: lib/ansible/modules/cloud/azure/azure_rm_webappslot.py [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+        # set location
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # get web app
+        webapp_response = self.get_webapp()
+
+        if not webapp_response:
+            self.fail("Web app {0} does not exist in resource grouop {1}.".format(self.webapp_name, self.resource_group))
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_webappslot.py]: group

File: lib/ansible/modules/cloud/azure/azure_rm_webappslot.py [hunk 3]
```diff
@@ -0,0 +1,1039 @@
...
+
+                    if self.frameworks[0]['name'] not in self.supported_linux_frameworks:
+                        self.fail('Unsupported framework {0} for Linux web app.'.format(self.frameworks[0]['name']))
+
+                    self.site_config['linux_fx_version'] = (self.frameworks[0]['name'] + '|' + self.frameworks[0]['version']).upper()
+
+                    if self.frameworks[0]['name'] == 'java':
+                        if self.frameworks[0]['version'] != '8':
+                            self.fail("Linux web app only supports java 8.")
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() != 'tomcat':
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_webappslot.py]: framwork_spec doesn't look like it requires `settings`, so potentially KeyErrors in several places here if not specified (probably should use `.get('settings', {})` or something similar, depending on usage.

**Review by nitzmahone — CHANGES_REQUESTED** [review]
Couple of minor things, but I can't think of a better way to lay out the UI given all the things it needs to do. The swap operation in particular will be nearly impossible to use in an idempotent fashion (at least I can't think of a way), so I guess this is one of those modules you just have to hope everything works right the first time ;)

**Comment by Fred-sun** [comment]
@yungezz

**Comment by Fred-sun** [comment]
@yungezz need you update

**Comment by Fred-sun** [comment]
@yungezz

File: lib/ansible/modules/cloud/azure/azure_rm_webappslot.py (hunk 3 above)
Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_webappslot.py]: fixed

File: lib/ansible/modules/cloud/azure/azure_rm_webappslot.py (hunk 2 above)
Comment (Fred-sun) [reply lib/ansible/modules/cloud/azure/azure_rm_webappslot.py]: @yungezz need you update

**Comment by yungezz** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@zikalino @nitzmahone @jborean93 Please help review this PR and push for merged when you're available! Thanks!

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**